partiql-types/src/main/kotlin/org/partiql/value/datetime/impl/OffsetTimestampHighPrecision.kt

```diff
@@ -188,8 +189,11 @@ internal class OffsetTimestampHighPrecision private constructor(
                         utc.hour, utc.minute, utc.decimalSecond,
                         timeZone
                     )
-                    is TimeZone.UtcOffset -> utc.plusMinutes(timeZone.totalOffsetMinutes.toLong())
-                        .let { of(it.year, it.month, it.day, it.hour, it.minute, it.decimalSecond, timeZone) }
+                    is TimeZone.UtcOffset -> {
+                        if (valueTimeZone == timeZone) this
```

Comment:
Many of these methods seem to create new instances. Is it alright to not follow the same here?

Reply:
It should be fine in my opinion. As the object is immutable.